File: greedy/1764_form_array_by_concatenating_subarrays_of_another_array/1764_form_array_by_concatenating_subarrays_of_another_array.py
from typing import List

class Solution:
    def canChoose(self, groups: List[List[int]], nums: List[int]) -> bool:
        i = 0
        n = len(nums)
        for group in groups:
            find = False
            while i + len(group) <= n:
                if group == nums[i:i+len(group)]:
                    find = True
                    i += len(group)
                    break
                else:
                    i += 1
            if not find:
                return False
        return True


groups = [[1,-1,-1],[3,-2,0]]
nums = [1,-1,0,1,-1,-1,3,-2,0]
res = Solution().canChoose(groups, nums)
print(res)

File: greedy/1764_form_array_by_concatenating_subarrays_of_another_array/1764_form_array_by_concatenating_subarrays_of_another_array.go
package main 

func canChoose(groups [][]int, nums []int) bool {
	i:=0
	j:=0
	for i<len(groups)&&j<len(nums){
		k:=0
		t:=j
		for k<len(groups[i])&&j<len(nums)&&groups[i][k]==nums[j]{
			k++
			j++
		}
		if k==len(groups[i]){
			i++
		}else {
			j = t+1
		}
	}
	return i==len(groups)
}
